CMakeLists.txt

```diff
@@ -70,6 +70,27 @@ include_directories(${CMAKE_BINARY_DIR}/include) # just to enable vscode intelli
 #set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native -mtune=native -flto") 
 add_definitions(-Wno-ignored-attributes) # to skip postgresql warnings.
 
+# Generate content type definition header file
+add_custom_target(
+    content_types ALL
+    DEPENDS "${CMAKE_BINARY_DIR}/include/li/http_server/content_types.hh"
+)
+
+add_custom_command(
+    OUTPUT "${CMAKE_BINARY_DIR}/include/li/http_server/content_types.hh"
+    COMMAND touch ${CMAKE_BINARY_DIR}/include/li/http_server/content_types.hh
+    COMMAND echo \"// This file is generated do not edit it.\" >> ${CMAKE_BINARY_DIR}/include/li/http_server/content_types.hh
```

Comment:
${CMAKE_BINARY_DIR}/include/li/* are symlinks to the ${CMAKE_SOURCE_DIR}/lithium/libraries/* . Can you write to ${CMAKE_SOURCE_DIR}/lithium/libraries/http_server/content_types.hh instead ? This would be easier to see that this is writing in the source dir.

Reply:
Will do.

Reply:
Yep, moved to ${CMAKE_SOURCE_DIR}/libraries/http_server/http_server/content_types.hh. Is this file path ok?

Reply:
yes thanks !